File: app/src/main/kotlin/org/translate/core/TranslateProviderFactory.java
package org.translate.core;


public class TranslateProviderFactory {

    public static TranslateProvider create(TranslateProvider.Provider provider, String from, String to){
        switch (provider){

            case YANDEX:
                return new YandexTranslateProvider(from, to);
            default:
                throw new RuntimeException("Couldn't create translate provider");
        }
    }
}

File: app/src/main/kotlin/org/book2words/translate/TranslateProviderFactory.kt
package org.book2words.translate

import org.book2words.data.CacheDictionary

public class TranslateProviderFactory {
    companion object {

        public fun create(provider: TranslateProvider.Provider, cache : CacheDictionary, from: String, to: String): TranslateProvider {
            when (provider) {

                TranslateProvider.Provider.YANDEX -> return YandexTranslateProvider(cache, from, to)
                else -> throw RuntimeException("Couldn't create translate provider")
            }
        }
    }
}
